x-pack/plugins/canvas/public/lib/loading_indicator.ts

```diff
@@ -9,6 +9,11 @@ import { loadingCount } from 'ui/chrome';
 
 let isActive = false;
 
+export interface LoadingIndicatorInterface {
```

Comment:
TS question: Shouldn't you tie this interface to the `loadingIndicator` object somehow?

Reply:
Typescript is able to infer that loadingIndicator conforms to this interface.

So, when we create the middleware, we have defined it should be given an argument called `loadingIndicator` that matches the shape defined in this interface.  The compiler is able to look at this `loadingIndicator`, or the dummy one that we make in the test file, and see that they conform.  

_I think_

Reply:
Yeah, I know it can infer types, which is why you normally don't need to provide a return type for functions. I thought maybe that also meant that you could import `loadingIndicator` as a type without the extra Interface definition, but I don't know. What you have it fine, just trying to learn.

> I think

😆